Add "Get Started with Bytebase" (#213)
bytebase/bytebase.com: content/docs/en/get-started-with-bytebase/run-an-sql-review-workflow-in-a-project.md

---
title: Run an SQL Review Workflow in a Project
---
## Create a project

> Developers manage their databases within the Bytebase Project. Therefore, before dealing with the sample databases, import them into a project first.  

Reviewer: ```suggestion
Developers manage their databases within the Bytebase Project. Therefore, before dealing with the sample databases, you need to transfer them into a project first.  
```

Reviewer: We don't use import because "import" is often used in "import data" and would cause confusion

Click `Projects` > `New Project` to create a project called "Employee".  

![_](/static/docs-assets/create-project.png)  

Choose **Standard**. The "Employee" project is created successfully and you are on its detail page.  

![_](/static/docs-assets/project-page.png)  

Choose `Transfer DB` to import the two databases synced from the connected instances. You can find them in the default project since they are not transferred into any specific project yet.  

![_](/static/docs-assets/transfer-db-1.png)  

![_](/static/docs-assets/transfer-db-2.png)  

Now the "employee" databases are available.  

## Add your teammates  

Remember to add your teammates to the workspace before you try to add them to a specific project.  

On the detail page of your created project, click `Settings` to add project members. In the below example, we add a project member "Adela".  

![_](/static/docs-assets/add-teammate.png)  

## Alter the database schema

> Now that everything is prepared, it's time to try the schema change workflow. The task is to add a column named "nickname" to the "employee" table. We will create a single pipeline to propagate the change from the Test instance to the Prod instance. By default, no manual approval is required on Test, so developers can run SQLs without DBAs' permission. Bytebase supports both UI and GitOps workflows to accomplish the task. This demo uses the UI workflow.  

Reviewer: ```suggestion
Now that everything is prepared, it's time to try the schema change workflow. The task is to add a column named "nickname" to the "employee" table. We will create a single pipeline to propagate the change from the `Test` instance to the `Prod` instance. By default, no manual approval is required on the `Test` environment, so developers can run SQLs without DBAs' approval. Bytebase supports both UI and GitOps workflows to accomplish the task. This demo uses the UI workflow.  
```

### Create an SQL issue

> The whole workflow begins with an SQL issue.  

Reviewer: ```suggestion
The whole workflow is encapsulated in a SQL issue.  
```

Open the "Employee" project. Click `Alter Schema`.  

![_](/static/docs-assets/alter-schema.png)  

Then a dialog box will pop up.  

![_](/static/docs-assets/alter-single-schema.png)  

> You need to choose `Alter multiple databases` to make the pipeline work.  

Reviewer: ```suggestion
You need to choose `Alter multiple databases` to create a pipeline propagating changes across multiple databases.  
```

![_](/static/docs-assets/alter-multiple-databases.png)  

You need to choose Alter multiple databases to make the pipeline work.  

![_](/static/docs-assets/issue-creation-page.png)  

> Fill in the title and the SQL field. Add a description if needed.  

Reviewer: ```suggestion
Fill in the required title, SQL statement, and an optional description.  
```

![_](/static/docs-assets/write-sql.png)  

> Click `Apply to other stages`. Assign this issue to a user for Prod review. Select yourself as the assignee for this demo, then click `Create`.  

Reviewer: ```suggestion
Click `Apply to other stages`.  For demo purposes, select yourself as the assignee, then click `Create`.  
```

![_](/static/docs-assets/create-issue.png)  

> The SQL Advisor kicks off immediately and checks your SQL statements automatically.  

Reviewer: ```suggestion
The SQL Advisor kicks off immediately and checks your SQL statements against various rules automatically.  
```

![_](/static/docs-assets/sql-advisor-2.png)  

> Your query runs successfully against the Test environment, which doesn't require manual approval, and waits for approval for the Prod environment.  

Reviewer: ```suggestion
The Test environment doesn't require manual approval, thus the SQL will be applied automatically after passing those advisor checks. Then you are waiting for approval for the Prod environment.  
```

### Review the SQL issue  

> Click `Approve` if everything is OK. Now you apply the change to the Prod environment.  

Reviewer: ```suggestion
Click `Approve` if everything is OK and the change will be applied to the Prod environment.  
```

![_](/static/docs-assets/approve-issue.png)  

![_](/static/docs-assets/approve-issue-2.png)  

> Click `Resovle issue`, and the issue will be done.  

Reviewer: ```suggestion
Now that we have applied the same SQL change to both the `Test` and `Prod` databases, we can click `Resolve issue` to finish the workflow.
```
